### PR TITLE
Osx app creation

### PR DESCRIPTION
**Pull request by pr0d1r2**
This is required in order to be in official homebrew repository: https://github.com/Homebrew/homebrew/pull/39987#discussion_r30893601

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -24,9 +24,20 @@
...
+    try:
+        import py2app
+    except ImportError as error:
+        print error
+        sys.exit(1)
+
 from distutils.core import setup
 from distutils.sysconfig import get_python_lib
 
+from subprocess import call
```
Comment (MrS0m30n3) [inline setup.py]: Why do you need the subprocess.call?

**Comment by MrS0m30n3** [comment]
What is the usage of the **youtube_dl_gui/icons/youtube-dl-gui_512x512.icns** file?

Also, i don't have a Mac to test the code but i can see that you manually copy the other files under **osx** directory (osx/Info.plist, osx/PkgInfo, osx/empty.lproj) https://github.com/Homebrew/homebrew/pull/39987/files#diff-8c3e3bae4209ea806fdd9b449a582314R20 at you formula. Don't you have to handle those files in the setup.py script or does the py2app creates them automatically?

**Comment by pr0d1r2** [comment]
Removed: `youtube_dl_gui/icons/youtube-dl-gui_512x512.icns` as it is contained in `osx/YoutubeDlGui.icns`

**Comment by MrS0m30n3** [comment]
Hi @pr0d1r2 

What about the other files under **osx** dir (Info.plist, PkgInfo, empty.lproj)?
Do we need them? Doesn't py2app creates them automatically?

**Comment by pr0d1r2** [comment]
Removed redundant files.

**Comment by MrS0m30n3** [comment]
Yeah, but since the **osx** directory contains only one file and it's an icon, you should remove the **osx** directory and move the *YoutubeDlGui.icns* file into *youtube_dl_gui/icons* or *youtube_dl_gui/icons/osx*.

 Also, don't forget to change the location of *YoutubeDlGui.icns* inside the setup.py file.

**Comment by pr0d1r2** [comment]
Done as suggested.

**Comment by MrS0m30n3** [comment]
:+1:

**Comment by pr0d1r2** [comment]
:smiley:

**Pull request merged**